### ISSUE TITLE
Make weighted_cross_entropy_with_logits consistent with sigmoid_cross_entropy_with_logits in signature

### ISSUE THREAD

**Issue by ppwwyyxx** [tensorflow/tensorflow#26337]
**System information**
- TensorFlow version (you are using): 1.13
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
Currently, the signature of the two functions are:
```python
@tf_export("nn.sigmoid_cross_entropy_with_logits")
def sigmoid_cross_entropy_with_logits(  # pylint: disable=invalid-name
    _sentinel=None,
    labels=None,
    logits=None,
    name=None):
```
And 
```python
@tf_export("nn.weighted_cross_entropy_with_logits")
def weighted_cross_entropy_with_logits(targets, logits, pos_weight, name=None):
```

TF2.0 is a good chance to make them consistent in the naming.

**Will this change the current api? How?**
Yes.

**Who will benefit with this feature?**
Users of the two functions will be less confused.

**Any Other info.**
A follow up of #7086

**Comment by yongtang** [comment]
Created a PR #26374 for the fix. Will let api review to decide if it is ok.

**Comment by ppwwyyxx** [comment]
```python
def sigmoid_cross_entropy_with_logits_v2(  # pylint: disable=invalid-name
    labels=None,
    logits=None,
    name=None):
```

```python
def weighted_cross_entropy_with_logits(targets, logits, pos_weight, name=None):
```

```python
@tf_export("nn.sparse_softmax_cross_entropy_with_logits")
def sparse_softmax_cross_entropy_with_logits(
    _sentinel=None,  # pylint: disable=invalid-name
    labels=None,
    logits=None,
    name=None):
```

There are still inconsistencies (targets vs labels; _sentinel in `sparse_softmax_cross_entropy_with_logits`).

**Comment by yongtang** [comment]
Added another PR #26380 for the remaining `sparse_softmax_cross_entropy_with_logits` and `weighted_cross_entropy_with_logits` issue.

**Comment by ppwwyyxx** [comment]
Thanks to both of you! 
Another small issue from @alextp's CL: after removing `_sentinel`, we should also remove the None default `labels=None, logits=None` from the signature. These meaningless defaults exist only because of sentinel.
